### PR TITLE
Add image settings to ofn-install

### PR DESCRIPTION
**Pull request by luisramos0**
#### What? Why?
To remove the page admin/image_settings/edit from OFN ([this PR](https://github.com/openfoodfoundation/openfoodnetwork/pull/5729)), we need to add an alternative way to change and track these settings.
Some are already in ofn-secrets:
s3_access_key
s3_secret
s3_images_bucket (which is converted to s3_bucket)

To keep consistency, in this PR, we add the 2 remaining S3 entries to ofn-secrets (to the example file) where the other ones are. It looks like this:
s3_headers: "{\"Cache-Control":"max-age=31557600\"}"
s3_protocol: "https"

In this PR we also set defaults for all image settings and copy custom settings from all servers to each server config.yml file.

#### What should we test?
I provisioned staging FR with this PR and I do get the images settings in application.yml as well as the custom ATTACHEMENT_PATH :+1: 

#### Release notes
Changelog Category: Changed
Image settings can now be set through ofn-install.

### CODE REVIEW

File: inventory/host_vars/www.openfoodnetwork.org.uk/config.yml
Comment (sauloperez) [inline inventory/host_vars/www.openfoodnetwork.org.uk/config.yml:28]: it's basically Katuma the one that doesn't use S3, right? In that case, I'd like to avoid that much duplication (we have quite a few already), which will ease adding new servers as well. We can just treat Katuma as the single exception and probably move it to S3 soon too. It's worth the simplification it will bring IMO.

tl;dr: I would make these settings the default value in roles/app/templates/application.yml.j2 (I believe all servers except Katuma share the same values) and use the var only if defined (just Katuma will).

Comment (luisramos0) [reply inventory/host_vars/www.openfoodnetwork.org.uk/config.yml:28]: I think we should stop using S3 and I dont think S3 should be the default. Relates to https://github.com/openfoodfoundation/openfoodnetwork/issues/5078

Anyway, it's a valid point, I'll fetch all the values from all the servers and see what's repeated. If all the servers have the same values, they should go to roles/app/templates/application.yml.j2 :+1:  I'd do this only to non-S3 specific settings though. I'll let you know when I have all the configs and then we decide, ok?

Comment (sauloperez) [reply inventory/host_vars/www.openfoodnetwork.org.uk/config.yml:28]: yes, treating Katuma's as an exception would indeed make things more manageable and easier to migrate from S3 to a better alternative.

Comment (luisramos0) [reply inventory/host_vars/www.openfoodnetwork.org.uk/config.yml]: ok, this is done in the new commits :+1: please have a look.

**Comment by luisramos0** [comment]
I am testing this with OFN PR that removes image_settings and it is not working. I need to investigate.
Meanwhile, can you have a look @sauloperez ?

File: roles/app/defaults/main.yml
Comment (sauloperez) [inline roles/app/defaults/main.yml:6]: why do write `u003E` instead of `>`? I fail to see why that would be problematic

File: inventory/host_vars/www.openfoodfrance.org/config.yml
Comment (sauloperez) [inline inventory/host_vars/www.openfoodfrance.org/config.yml:28]: This overriden path seems a bit odd. It includes the default `public/spree/products/:id/:style/:basename.:extension` path but prepends it with a relative path from the home folder. I'm not sure it would work. Can't we simply remove it so the default one applies?

File: roles/app/templates/application.yml.j2
Comment (sauloperez) [inline roles/app/templates/application.yml.j2:34]: this is fine as long as we change Katuma's secrets to use `s3_backups_region` instead as that it's intended use (I'll do that now).

Comment (sauloperez) [reply roles/app/templates/application.yml.j2:34]: Done in https://github.com/openfoodfoundation/ofn-secrets/pull/42

File: roles/app/defaults/main.yml
Comment (luisramos0) [reply roles/app/defaults/main.yml:6]: yeah, I dont know, I just copied what was on the DB. Maybe I can change to > here and it will be converted when saving to db :+1:

File: inventory/host_vars/www.openfoodfrance.org/config.yml
Comment (luisramos0) [reply inventory/host_vars/www.openfoodfrance.org/config.yml:28]: this is copied from the live DB. I thought you have done this :-) maybe @Matt-Yorkley ?

File: roles/app/defaults/main.yml
Comment (sauloperez) [reply roles/app/defaults/main.yml:6]: yes, let's just try and see if we can avoid some of this obfuscation.

File: inventory/host_vars/www.openfoodfrance.org/config.yml
Comment (sauloperez) [reply inventory/host_vars/www.openfoodfrance.org/config.yml:28]: we should give it a try

Comment (Matt-Yorkley) [reply inventory/host_vars/www.openfoodfrance.org/config.yml:28]: Different production servers have different settings here, there's not a single value we can apply to all of them. It's about what setting was used when the server first started using S3, which wasn't always the default. All images for that server will now be stored with the given path. A couple of servers needed a different path prefix after the changes to symlinking in the deployment process...

File: roles/app/defaults/main.yml
Comment (Matt-Yorkley) [inline roles/app/defaults/main.yml:2]: The build is failing due to this line. I think we either need to escape all of the double quotes, or wrap in single quotes:
```yaml
"{\"Cache-Control\":\"max-age=31557600\"}"
```
```yaml
'{"Cache-Control":"max-age=31557600"}'
```

**Comment by Matt-Yorkley** [comment]
The one useful thing in that page that I guess we will now be missing was a button for a "regenerate thumbnails" action. If any changes are made to the scaling / sizing settings (with the weird > and # symbols for example) the images need to be regenerated. I guess it's not a big deal though.

**Comment by sauloperez** [comment]
Let's copy that feature in a rake task then :ok_hand:

File: roles/app/defaults/main.yml
Comment (luisramos0) [reply roles/app/defaults/main.yml:6]: This does get to the DB, in DE we have > in AU we have the encoded version.
I removed this now. We can verify while testing the PR.

File: inventory/host_vars/www.openfoodfrance.org/config.yml
Comment (luisramos0) [reply inventory/host_vars/www.openfoodfrance.org/config.yml:28]: I am not sure why we need the full path here but I am going to keep it as is, it's not really the scope of this PR.

File: roles/app/defaults/main.yml
Comment (luisramos0) [reply roles/app/defaults/main.yml]: :+1: thanks, fixed.

**Comment by luisramos0** [comment]
@Matt-Yorkley the regenerate thumbnails was a message to the user:
`image_settings_warning: "You will need to regenerate thumbnails if you update the paperclip styles. Use rake paperclip:refresh:thumbnails CLASS=Spree::Image to do this."
`I have added a comment to config example so users will know what they need to do if they change the attachment styles.

**Comment by luisramos0** [comment]
This is ready for approval I think.
I have tested it with https://github.com/openfoodfoundation/openfoodnetwork/pull/5729 and it does work, the settings do get to the DB as expected :+1:

**Comment by luisramos0** [comment]
Here we are keeping all settings as they were, all changes can be done at another time, issue for that created here: https://github.com/openfoodfoundation/openfoodnetwork/issues/5911

**Review by sauloperez — APPROVED** [review]

**Comment by luisramos0** [comment]
rebased to resolve conflicts.

This needs a second review.

File: inventory/host_vars/_example.com/secrets.example.yml
Comment (mkllnk) [inline inventory/host_vars/_example.com/secrets.example.yml:47]: I think that this will fail because `application.yml.j2` doesn't have `if ... defined` guard clauses for these.

File: roles/app/defaults/main.yml
Comment (mkllnk) [inline roles/app/defaults/main.yml:6]: The small `100x100#` setting is actually a bug. It's the Spree default but OFN's design expects `112x112#` for the product listing. We are probably going to change this for the mobile design though.

**Review by mkllnk — CHANGES_REQUESTED** [review]
I think we need to fix the Stripe config. Apart from that it's looking good. Well, I would change the default small image size to help new instances until we fix the problem in the mobile ux update.

File: inventory/host_vars/_example.com/config.yml
Comment (mkllnk) [inline inventory/host_vars/_example.com/config.yml:34]: :bomb: I just read this is in the docs:

> WARNING: regeneration is broken on S3 and will truncate your original images.
https://github.com/thoughtbot/paperclip/wiki/Thumbnail-Generation#generatingregenerating-your-thumbnails

That is true for version 6.1.0 while we use version 3.4.2. I don't know when that bug was introduced but the issue was reported nine months after the last release. So we might be safe for a while.

File: inventory/host_vars/_example.com/secrets.example.yml
Comment (luisramos0) [reply inventory/host_vars/_example.com/secrets.example.yml:47]: I think this is ok because they are defined as empty in roles/app/defaults/main.yml 
I just tested it in staging katuma by removing all stripe keys from the secrets file, this is how it ends up in the server:
image
:+1:

File: roles/app/defaults/main.yml
Comment (luisramos0) [reply roles/app/defaults/main.yml:6]: :+1:  agree, here I am just copying what's on the DB to the config file. Fixing this is out of scope for this PR. I created an issue for that: https://github.com/openfoodfoundation/openfoodnetwork/issues/5911

File: inventory/host_vars/_example.com/config.yml
Comment (luisramos0) [reply inventory/host_vars/_example.com/config.yml:34]: it's good to know :+1:

**Comment by luisramos0** [comment]
Hi @mkllnk I think the PR is ok as is, I tested in Katuma staging with no stripe secrets defined and it worked.

**Review by mkllnk — APPROVED** [review]

**Pull request merged**
